### PR TITLE
[datadog_cloud_configuration_rule] Added filter field

### PR DESCRIPTION
**Pull request by symphony-elias**
https://datadoghq.atlassian.net/browse/SEC-9912

### CODE REVIEW

File: docs/resources/cloud_configuration_rule.md [hunk 1]
```diff
@@ -66,6 +66,7 @@ resource "datadog_cloud_configuration_rule" "myrule" {
 
 ### Optional
 
+- `filter` (Block List) Additional queries to filter matched events before they are processed. Defaults to empty list (see [below for nested schema](#nestedblock--filter))
```
Comment (brett0000FF) [inline docs/resources/cloud_configuration_rule.md:69]:
```suggestion
- `filter` (Block List) Additional queries to filter matched events before they are processed. Defaults to empty list (see [nested schema](#nested-schema-for-filter)).
```

File: docs/resources/cloud_configuration_rule.md [hunk 2]
```diff
@@ -75,6 +76,14 @@ resource "datadog_cloud_configuration_rule" "myrule" {
 
 - `id` (String) The ID of this resource.
 
+<a id="nestedblock--filter"></a>
+### Nested Schema for `filter`
```
Comment (brett0000FF) [inline docs/resources/cloud_configuration_rule.md:80]:
```suggestion
### Nested Schema for filter
```

File: docs/resources/cloud_configuration_rule.md [hunk 3]
```diff
@@ -75,6 +76,14 @@ resource "datadog_cloud_configuration_rule" "myrule" {
...
+
+Required:
+
+- `action` (String) The type of filtering action. Valid values are `require`, `suppress`.
```
Comment (brett0000FF) [inline docs/resources/cloud_configuration_rule.md:84]:
```suggestion
- `action` (String) The type of filtering action. Valid values are `require` or `suppress`.
```

**Review by brett0000FF — APPROVED** [review]
LGTM. Thanks! I offered a few small suggestions.

Also, I wanted to note that our style prefers sentence-case headings for documentation. I didn't change that here since the PR has a smaller scope.

**Comment by symphony-elias** [comment]
@brett0000FF Thanks for your suggestions. However, the documentation is automatically generated and CI fails if there is any discrepancy between the documentation on the branch and the generated one. Please contact @DataDog/integrations-tools-and-libraries team if you have suggestions to make regarding the way the documentation is generated

**Review by skarimo — APPROVED** [review]

File: docs/resources/cloud_configuration_rule.md (hunk 3 above)
Comment (skarimo) [reply docs/resources/cloud_configuration_rule.md:84]: same as above

**Pull request merged**